### DISCUSSION TITLE
How do you use the state property of navigation?

### DISCUSSION THREAD

**Discussion by geoff-harper** [TanStack/router#2324] [Q&A]
The `ToOptions` interface is as follows

```ts
type ToOptions = {
  from?: ValidRoutePath | string
  to?: ValidRoutePath | string
  hash?: true | string | ((prev?: string) => string)
  state?: true | HistoryState | ((prev: HistoryState) => HistoryState)
} & SearchParamOptions &
  PathParamOptions

type SearchParamOptions = {
  search?: true | TToSearch | ((prev: TFromSearch) => TToSearch)
}

type PathParamOptions = {
  path?: true | Record<string, TPathParam> | ((prev: TFromParams) => TToParams)
}
```

I want to pass a list of 100+ ids to a new route. I'd prefer to do this through my router since it's cleaner but it isn't feasible to put them in the search string. The `state` property seems ideal for this but I can't find any other documentation on it other than in this interface. If this isn't the way to go, is there a better way? I'd prefer to avoid temporarily stashing it in some non-router global state to bridge it between the new routes since that feels like a hacky work around

**Comment by SeanCassiere** [comment]
This has been answered here: https://github.com/TanStack/router/discussions/1342#discussioncomment-8848490
